File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    num=dict()
    for p in participant:
        num[p]=num.get(p,0)+1
    for c in completion:
        num[c]-=1
        if num[c]==0:
            del num[c]
    return list(num.keys()).pop()

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> check = new HashMap<>();
        for (int i = 0; i < participant.length; i ++) {
            if (!check.containsKey(participant[i])) {
                check.put(participant[i], 1);
            } else {
                check.put(participant[i], check.get(participant[i]) + 1);
            }
        }
        for (int i = 0; i < completion.length; i ++) {
            check.put(completion[i], check.get(completion[i]) - 1);
        }
        for (int i = 0; i < participant.length; i ++) {
            if (check.get(participant[i]) > 0) {
                answer = participant[i];
                break;
            }
        }
        return answer;
    }
}